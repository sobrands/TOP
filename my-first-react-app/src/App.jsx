import { useState } from "react";
import "./App.css";

function Button({
  text = "Click Me!",
  color = "blue",
  fontSize = 12,
  handleClick,
}) {
  const buttonStyle = {
    color: color,
    fontSize: fontSize + "px",
  };
  return (
    <button style={buttonStyle} onClick={handleClick}>
      {text}
    </button>
  );
}

function App() {
  const [heading, setHeading] = useState("Our first test");
  const handleButtonClick = () => {
    setHeading("Our second test now");
  };

  return (
    <div>
      <h1>{heading}</h1>
      <Button handleClick={handleButtonClick} />
    </div>
  );
}

export default App;
