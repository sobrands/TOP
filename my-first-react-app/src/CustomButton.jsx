export default function CustomButton({ onClick }) {
  return <button onClick={onClick}>Click Me!</button>;
}
