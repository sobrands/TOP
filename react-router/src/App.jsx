import "./App.css";
import { Link } from "react-router-dom";

function App() {
  return (
    <>
      <h1>Hello from the main page of the app!</h1>
      <p>Here are some examples of links to other pages</p>
      <nav>
        <ul>
          <li>
            <Link to="profile">Profile page</Link>
          </li>
        </ul>
      </nav>
    </>
  );
}

export default App;
