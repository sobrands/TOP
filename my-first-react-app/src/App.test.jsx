import { expect, describe, it, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import CustomButton from "./CustomButton";

describe("App component", () => {
  it("renders our first test", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders our second test after button click", async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", { name: "Click Me!" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(
      /our second test now/i,
    );
  });
});

describe("Custom Button", () => {
  it("should render a button with the text 'Click me'", () => {
    render(<CustomButton onClick={() => {}} />);

    const button = screen.getByRole("button", { name: "Click Me!" });

    expect(button).toBeInTheDocument();
  });

  it("should call the onClick function when clicked", async () => {
    const onClick = vi.fn();
    const user = userEvent.setup();
    render(<CustomButton onClick={onClick} />);

    const button = screen.getByRole("button", { name: "Click Me!" });

    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it("should not call the onClick function when it isn't clicked", async () => {
    const onClick = vi.fn();
    render(<CustomButton onClick={onClick} />);

    expect(onClick).not.toHaveBeenCalled();
  });
});
