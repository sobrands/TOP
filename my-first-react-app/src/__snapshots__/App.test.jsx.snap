// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders our first test 1`] = `
<div>
  <div>
    <h1>
      Our first test
    </h1>
    <button
      style="color: blue; font-size: 12px;"
    >
      Click Me!
    </button>
  </div>
</div>
`;
